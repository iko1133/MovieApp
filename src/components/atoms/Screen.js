import React from "react";
import { Platform } from "react-native";
import { StatusBar } from "expo-status-bar";
import Constants from "expo-constants";
import styled from "styled-components/native";

const Background = styled.KeyboardAvoidingView`
  background-color: ${({ theme }) => theme.BACKGROUND_COLOR};
  height: 100%;
  justify-content: center;
`;

const SafeAreaView = styled.SafeAreaView`
  flex: 1;
  padding-top: ${Constants.statusBarHeight}px;
`;

const InnerContainer = styled.View`
  flex: 1;
`;

/**
 * A screen component, accounting for the most page needs to be wrapped in (like KeyboardAvoidingView and SafeAreaView)
 * @param {object} children - children object to be rendered under screen
 * @param {object} style - styling of the inner container
 */

const Screen = ({ children, style }) => {
  return (
    <Background behavior={Platform.OS === "ios" ? "padding" : "height"}>
      <SafeAreaView>
        <InnerContainer style={style}>{children}</InnerContainer>
        <StatusBar style={"dark"} />
      </SafeAreaView>
    </Background>
  );
};

export default Screen;
