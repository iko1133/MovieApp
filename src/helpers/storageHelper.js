import AsyncStorage from "@react-native-async-storage/async-storage";

/**
 * A function that will save given object under given key
 * @param {object} object - An object that needs to be saved in storage
 * @param {string} key - A string key the object should be saved under
 * @param {string} errorText - A string that will be displayed if something goes wrong
 */
export const storeObject = async (
  object,
  key,
  errorText = "There was an error while trying to save"
) => {
  try {
    const jsonObj = JSON.stringify(object);
    await AsyncStorage.setItem(key, jsonObj);
  } catch (e) {
    alert(errorText);
  }
};

/**
 * A function that will return an object saved under given key or null if none was found
 * @param {string} key - A string key that desired object should be saved under
 * @param {string} errorText - A string that will be displayed if something goes wrong
 * @returns Object under given key or null if none was found
 */
export const getObject = async (key, errorText) => {
  try {
    const jsonVal = await AsyncStorage.getItem(key);
    return jsonVal != null ? JSON.parse(jsonVal) : null;
  } catch (e) {
    alert(errorText);
  }
};
