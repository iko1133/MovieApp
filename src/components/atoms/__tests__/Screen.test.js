import React from "react";
import renderer from "react-test-renderer";
import styled from "styled-components/native";

import { WrapInThemeProvider } from "helpers/testsHelper";
import Screen from "../Screen";

const RandomText = styled.Text`
  font-size: 16px;
`;

const Button = styled.TouchableOpacity`
  background-color: red;
  height: 100%;
  width: 100%;
`;

describe("Testing Screen", () => {
  it("Renders Screen", () =>
    expect(renderer.create(WrapInThemeProvider(<Screen />))).toMatchSnapshot());

  it("Renders Screen with text in it", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(
          <Screen>
            <RandomText>Random Text!</RandomText>
          </Screen>
        )
      )
    ).toMatchSnapshot());

  it("Renders Screen with button in it", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(
          <Screen>
            <Button />
          </Screen>
        )
      )
    ).toMatchSnapshot());
});
