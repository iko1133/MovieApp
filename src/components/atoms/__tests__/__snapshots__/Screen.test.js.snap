// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Screen Renders Screen 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#F7F1E9",
          "height": "100%",
          "justifyContent": "center",
        },
      ],
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingTop": 20,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    />
  </RCTSafeAreaView>
</View>
`;

exports[`Testing Screen Renders Screen with button in it 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#F7F1E9",
          "height": "100%",
          "justifyContent": "center",
        },
      ],
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingTop": 20,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "red",
            "height": "100%",
            "opacity": 1,
            "width": "100%",
          }
        }
      />
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`Testing Screen Renders Screen with text in it 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#F7F1E9",
          "height": "100%",
          "justifyContent": "center",
        },
      ],
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingTop": 20,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
            },
          ]
        }
      >
        Random Text!
      </Text>
    </View>
  </RCTSafeAreaView>
</View>
`;
